### PR TITLE
[WIP] Improve explained variance visualizer 

### LINKED ISSUE #316: Finish decomposition explained variance visualizer
Thanks to @georgerichardson we have an explained variance visualizer for PCA decomposition. This needs to be wrapped up as follows:

- [ ] expressiveness for other explained variances 
- [ ] labeling and key values features
- [ ] tests 
- [ ] documentation 

See #238

### PR DESCRIPTION
**Pull request by RaulPL**
Fixes #316 

Initial work to finish explained variance visualizer 

Changes proposed in this pull request:
- Added a few docs to the class
- I'm returning a bar plot because I think its easier to visualize

I am really new to this so I don't know what kind of tests can I program for this class.

@DistrictDataLabs/team-oz-maintainers

### CODE REVIEW

**Comment by lwgray** [comment]
Thank you for joining us for the Yellowbrick Sprint at PyCon Cleveland. And also, thank you for submitting your first PullRequest - oh yeah!

We will review this asap. We have been quite busy with all the great contributions from sprinters like yourself.  However, feel free to work on another issue.  We hope you continue to contribute.

File: yellowbrick/features/decomposition.py [hunk 1]
```diff
@@ -117,8 +131,14 @@ def transform(self, X):
         return self.pca_features
 
     def draw(self):
+        if self.ax is None:
+            self.ax = self.gca()
```
Comment (bbengfort) [inline yellowbrick/features/decomposition.py]: Lines 134 and 135 are no longer necessary, the `Visualizer.ax` property ensures that if `self.ax` is None, then either the current axes object is fetched or a new axes is created.

File: yellowbrick/features/decomposition.py [hunk 2]
```diff
@@ -91,16 +107,14 @@ class ExplainedVariance(FeatureVisualizer):
...
         self.colormap = colormap
         self.n_components = n_components
         self.center = center
         self.scale = scale
-        self.pipeline = Pipeline([('scale', StandardScaler(with_mean=self.center,
-                                                           with_std=self.scale)), 
-                                  ('pca', PCA(n_components=self.n_components))])
+        self.pipeline = Pipeline(
+            [('scale', StandardScaler(with_mean=self.center, with_std=self.scale)),
+             ('pca', PCA(n_components=self.n_components))])
```
Comment (bbengfort) [inline yellowbrick/features/decomposition.py]: What happens here if `n_components` is None?

File: yellowbrick/features/decomposition.py [hunk 3]
```diff
@@ -117,8 +131,14 @@ def transform(self, X):
         return self.pca_features
 
     def draw(self):
+        if self.ax is None:
+            self.ax = self.gca()
         X = self.explained_variance_
-        self.ax.plot(X)
+        if self.n_components:
```
Comment (bbengfort) [inline yellowbrick/features/decomposition.py]: Great catch!

**Comment by bbengfort** [comment]
@RaulPL let me know when you're ready for me to review!

**Comment by bbengfort** [comment]
@RaulPL just saw the proj_features parameter - would it be possible for you to divide this into two branches/pull requests - one for the explained variance improvements and one for the biplot behavior?

**Comment by rebeccabilbro** [comment]
Hey there @RaulPL  - since the conversation has gone a bit stale here, I'm going to go ahead and close this one. Please feel free to open up a new PR as soon as you're ready to get back into this. We'd love to have you contribute again (P.S. thanks for the biplots!! They're in the latest release!)

**Pull request closed without merging**